components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/proxy/NetworkingConfigurator.java

```diff
@@ -0,0 +1,57 @@
+package com.gradle.enterprise.proxy;
+
+import com.gradle.enterprise.cli.ConsoleLogger;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.Properties;
+
+public class NetworkingConfigurator {
+
+    public static void configureNetworking(Optional<Path> networkSettingsFile, ConsoleLogger logger) {
```

Comment:
I find an Optional as an arg not ideal. Instead, I would move that logic to the caller with something like networkSettingsFile.ifPresent(configure(it, logger)).

Reply:
✅ Done